cdap-data-fabric/src/main/java/co/cask/cdap/data/view/ViewAdmin.java

```diff
@@ -42,19 +42,26 @@ public ViewAdmin(ViewStore store, ExploreFacade explore, ExploreTableNaming nami
   }
 
   public boolean createOrUpdate(Id.Stream.View viewId, ViewSpecification spec) throws Exception {
-    if (store.exists(viewId)) {
-      explore.disableExploreStream(viewId.getStream(), naming.getTableName(viewId));
+    try {
+      ViewSpecification previousSpec = store.get(viewId);
+      if (previousSpec == null) {
```

Comment:
Asking because it seems like it will always throw `NotFoundException` and the `store.get` is not nullable.